### ISSUE TITLE
Ryujinx creates new input device when wired controller disconnects

### ISSUE THREAD

**Issue by Itamai** [Ryujinx/Ryujinx#3048]
## Bug Report

![image](https://user-images.githubusercontent.com/53093696/151412456-170edc3b-9338-4b66-a21e-b80b34219a1e.png)

### What's the issue you encountered?

My cable wire has a slight break and sometimes little moves are enough to disconnect the controller from my pc and every time this happens I noticed that my controller doesnt work anymore because it creates a new input device in the controller settings. So i need to select the new created input device to be able to play again

### How can the issue be reproduced?

Let the controller disconnect and connect itself while ryujinx is open and it should generate a new input device.
Edit: after going through some commits I found out that the issue started happening after this PR #2553

### Log file

Not needed

### Environment?

- Ryujinx version: 1.1.10</br>

**Comment by Squall-Leonhart** [comment]
Updating to 2.0.18+ has a significant change in that Windows.Gaming.Input is used instead of Xinput.

**Comment by TurinTur** [comment]
Yeah, I have two controllers, and they work perfectly in Cemu, Yuzu, or any normal Steam games. But one of them it has cuts of few ms (I guess, as I never notice it!) and it causes for the controller to appear twice on the list, and even worse, for the controllers to disconnect, I have to select them again on the list.

**Comment by Itamai** [comment]
Update: I did some testing myself and added some breakpoints in the `SDL2GamepadDriver.cs` file and it seems like sometimes it doesn't call the "HandleJoyStickDisconnected" event.

I'm not sure, but if you selected a different device than the disconnecting one, the event shouldn't fire.
My guess is that for example if you have device id 0 selected but your gamepad is assigned to the device id 1
then the events expect the disconnect to happen at id 1 but because id 0 is selected it doesn't trigger the disconnect event.

**Comment by MetrosexualGarbodor** [comment]
Might be fixed on #3317 or #2507, or both.

**Comment by Itamai** [comment]
> Might be fixed on #3317 or #2507, or both.
I will test both PRs this weekend and post an update here or on the PR. Thank you for letting me know.

**Comment by Itamai** [comment]
Update: I can no longer reproduce the problem tested on latest main build. 
Could it have been a problem with Windows itself?
I'm in the Windows 11 insider program and we've had about 3-4 new build updates since I wrote the update comment 23 days ago.
https://blogs.windows.com/windows-insider/posts/

So this issue could potentially happen to anyone who has lower Windows 11 build version than 22598. Unfortunately, I can't confirm my statement and test the older builds, since we can only access the previous build used with the insider program.
Back then I was able to reproduce this issue on multiple builds with no issues (main, vulkan)
